src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -75,6 +77,35 @@ public static string CreateWorkItemTitle(this Run run)
                     (locationName == null ? "" : " (in " + locationName + ")");
         }
 
+        public static Dictionary<string, int> FetchToolResultCounts(this SarifLog log)
+        {
+            if (log == null) { throw new NullReferenceException(); }
+            if (log.Runs == null) { throw new ArgumentNullException(nameof(log.Runs)); }
+
+            Dictionary<string, int> resultCountsByTool = new Dictionary<string, int>();
+            int appropriateResultCount = 0;
+
+            foreach (Run run in log?.Runs)
+            {
+                if (run != null)
+                {
+                    if (run?.Tool?.Driver?.Name != null && run?.Results != null)
+                    {
+                        foreach (Result result in run?.Results)
+                        {
+                            if (result.AppropriateForFiling())
```

Comment:
>AppropriateForFiling [](start = 39, length = 20)

Suggest removing this filter. The general helper should just provide the actual data. That means any caller needs to have filtered out what data they aren't interested. Our filer will have completed this work before your helper gets called.